assets/src/block-patterns/translate.js

```diff
@@ -0,0 +1,103 @@
+import { _x } from '@wordpress/i18n';
+import { isEmpty } from 'lodash';
+
+function isObject( object ) {
+  return object !== null && typeof object === 'object';
+}
+
+/**
+ * Copy/paste of https://github.com/WordPress/gutenberg/blob/58c78ec5200a38401281c14186095bced2f6f12a/packages/blocks/src/api/registration.js#L296
+ *
+ * Translates block settings provided with metadata using the i18n schema.
+ *
+ * @param {string|string[]|Object[]} i18nSchema   I18n schema for the block setting.
+ * @param {string|string[]|Object[]} settingValue Value for the block setting.
+ * @param {string}                   textdomain   Textdomain to use with translations.
+ *
+ * @return {string|string[]|Object[]} Translated setting.
+ */
+function translateBlockSettingUsingI18nSchema(
+  i18nSchema,
+  settingValue,
+  textdomain
+) {
+  if ( typeof i18nSchema === 'string' && typeof settingValue === 'string' ) {
+    return _x( settingValue, i18nSchema, textdomain );
+  }
+  if (
+    Array.isArray( i18nSchema ) &&
+    ! isEmpty( i18nSchema ) &&
```

Comment:
Just a small comment, instead of invoking a lodash function you can check `i18nSchema` content through `i18nSchema.length` because you've already know that `i18nSchema` it's an array. Does it make sense?

Reply:
Yes :+1: